pkg/signature/kms/azure/client.go

```diff
@@ -268,14 +253,30 @@ func (a *azureVaultClient) getKey(ctx context.Context) (keyvault.KeyBundle, erro
 	return key, err
 }
 
-func (a *azureVaultClient) public() (crypto.PublicKey, error) {
-	return a.keyCache.Get(cacheKey)
+func (a *azureVaultClient) public(ctx context.Context) (crypto.PublicKey, error) {
+	var lerr error
+	loader := ttlcache.LoaderFunc[string, crypto.PublicKey](
```

Comment:
Same here, should we set the loader on initialization rather than on each access?

Reply:
A global loader is possible, but would require explicit interfaces to propagate errors properly.  The upstream author has elaborated on this in https://github.com/jellydator/ttlcache/issues/74#issuecomment-1133012806 (btw, the whole conversation on that issue is a very good read because it explains the motivation for separating the Loader from the Cache interface).

In fact, I've first implemented in with the version bump that way, but realized that it adds a lot of boiler code without much benefit. The implementation I'm promposing here actually removes code, improving readability and error propagation.